lib/src/content_type.dart

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+
+import 'package:http_parser/http_parser.dart';
+
+/// Returns the [Encoding] that corresponds to [charset].
+///
+/// Returns [fallback] if [charset] is null or if no [Encoding] was found that
+/// corresponds to [charset].
+Encoding encodingForCharset(String charset, [Encoding fallback = LATIN1]) {
+  if (charset == null) return fallback;
+  var encoding = Encoding.getByName(charset);
+  return encoding == null ? fallback : encoding;
+}
+
+/// Determines the encoding from the media [type].
+///
+/// Returns [fallback] if the charset is not specified in the [type] or if no
+/// [Encoding] was found that corresponds to the `charset`.
+Encoding encodingForMediaType(MediaType type, [Encoding fallback = LATIN1]) {
+  if (type == null) return fallback;
+  return encodingForCharset(type.parameters['charset'], fallback);
+}
+
+/// Modifies the media [type]'s [encoding].
+MediaType modifyEncoding(MediaType type, Encoding encoding) =>
```

Comment:
This doesn't seem to be used anywhere, and I'm not sure it would be worth factoring out even if it were...

Reply:
I was going to use it when creating the `content-type` header from a body and also its in the multipart stuff.

Reply:
It just doesn't seem like it provides substantial value over just inlining the definition.